roles/matrix-backup-borg/tasks/setup_install.yml

```diff
@@ -1,4 +1,7 @@
 ---
+- import_tasks: "{{ role_path }}/tasks/util/detect_existing_postgres_version.yml"
```

Comment:
Instead of duplicating this detection code, why not not import `roles/matrix-postgres/tasks/util/detect_existing_postgres_version.yml`?

I don't know why the `matrix-postgres-backup` role duplicates the detection code as well. Perhaps it was easier to do? Perhaps including from the `matrix-postgres` role did not work or some changes are required to make it work?

Supposing that it works, after this `import_tasks` call, instead of  accepting all results, you may add another `fail` task for versions that you don't support.

```yaml
- block:
  - import_tasks: "{{ role_path }}/../matrix-postgres/tasks/util/detect_existing_postgres_version.yml"
    when: matrix_backup_borg_postgresql_enabled|bool
  
  - name: Fail if detected Postgres version is unsupported
    fail:
      msg: "You cannot use borg backup with such an old version ({{ matrix_postgres_detected_version }}) of Postgres. Consider upgrading - link to docs for upgrading Postgres here"
    when: "matrix_postgres_detected_version not in matrix_backup_borg_supported_postgres_versions"
  
  - name: Set the correct borg backup version to use
    set_fact:
       matrix_backup_borg_version: "{{ matrix_postgres_detected_version }}"
  when: "matrix_backup_borg_version != '' and matrix_backup_borg_postgresql_databases_hostname == 'matrix-postgres'"

# Do something for when matrix_backup_borg_postgresql_databases_hostname is not 'matrix-postgres' (external Postgres server).
```

`matrix_backup_borg_supported_postgres_versions` can be defined as `['12', '13', `'14']` in `roles/matrix-backup-borg/defaults/main.yml`.

Reply:
I changed a bit your suggestion, because it doesn't matter if postgres is self-hosted or not for backups